### ISSUE TITLE
Create integration test env that handles auth protected routes correctly

### ISSUE THREAD

**Issue by Bouncey** [FreeCodeCamp/open-api#81]

**Comment by ojongerius** [comment]
I started looking at this out of interest. It looks like we can easily created a mocked context, and pass it in to the query like so:

```js
   // Create mock context
  const mockedContext = {
    headers: {
    'Content-Type': 'application/json',
    authorization: 'Bearer ' + token
    }
  };

  const rootValue = {};
  // Pass context into they query
  const result = await graphql(graphqlSchema, query, rootValue, mockedContext);
```

This will make tests return correct responses for non existent and malformed tokens, still have to test against a correct token.

To do that, we create a random secret to use, have the server use that for the verification, and create the tokens using the same secret.

When that works we can make it reusable across tests, using the Jest global environment setup for Mongo.

**Comment by ojongerius** [comment]
Ha! Just had lunch and creating a valid token really is as simple as:

```js
const token = jwt.sign({ id: 123, name: 'Charlie' }, JWT_CERT);
```

I'll factor this so that it's available in the test environment and raise a PR.

**Comment by ojongerius** [comment]
Changes from #83 were merged in via #58
